### DISCUSSION TITLE
watchOS Setup

### DISCUSSION THREAD

**Discussion by azlekov** [JuulLabs/kable#320] [Q&A]
Hello 👋,

I was thinking if Kable support watchOS target as well? 
What do I have to change or adjust something in my project targets or configuration in order to make it work on watchOS or it should work out of the box?

Thanks!

**Comment by twyatt** [comment]
It is my understanding that watchOS utilizes the same Core Bluetooth that macOS does. If so, then it should be as easy as adding the necessary targets — then it _should_ just work. 🤞 

I'm not exactly sure which [targets](https://kotlinlang.org/docs/multiplatform-dsl-reference.html#targets) would be necessary, but the available watchOS ones are:

- `watchosArm32`
- `watchosArm64`
- `watchosX86`
- `watchosX64`
- `watchosSimulatorArm64`

Then you'd just configure the main and test source sets for the targets that were added, similar to [how macOS was done](https://github.com/JuulLabs/kable/blob/4240f99e407c164004860ecb618bff22040fc529/core/build.gradle.kts#L79-L85).

**Reply by azlekov** [reply]
Do you think it's a good idea to document this in the readme or somewhere else? 

I can help once I confirm how to configure this.

**Reply by twyatt** [reply]
> Do you think it's a good idea to document this in the readme or somewhere else?

I don't think this is necessary, as it'd be preferable that these targets are added (rather than documented how to add them).
The main reason they haven't been added already is being limited on time to try/test them.

PRs are welcomed/appreciated for adding these targets. If you have devices to test with, then it would be awesome if you tested some of the above mentioned targets and create PR(s).
